chia/wallet/coin_selection.py

```diff
@@ -0,0 +1,160 @@
+import logging
+import random
+from typing import Dict, List, Optional, Set
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint64, uint128
+from chia.wallet.wallet_coin_record import WalletCoinRecord
+
+
+async def select_coins(
+    spendable_amount: uint128,
+    max_coin_amount: int,
+    spendable_coins: List[WalletCoinRecord],
+    unconfirmed_removals: Dict[bytes32, Coin],
+    log: logging.Logger,
+    amount: uint128,
+    exclude: Optional[List[Coin]] = None,
+    min_coin_amount: Optional[uint128] = None,
+) -> Set[Coin]:
+    """
+    Returns a set of coins that can be used for generating a new transaction.
+    """
+    if exclude is None:
+        exclude = []
+    if min_coin_amount is None:
+        min_coin_amount = uint128(0)
+
+    if amount > spendable_amount:
+        error_msg = (
+            f"Can't select amount higher than our spendable balance.  Amount: {amount}, spendable: {spendable_amount}"
+        )
+        log.warning(error_msg)
+        raise ValueError(error_msg)
+
+    log.debug(f"About to select coins for amount {amount}")
+
+    max_num_coins = 500
+    sum_spendable_coins = 0
+    valid_spendable_coins: List[Coin] = []
+
+    for coin_record in spendable_coins:  # remove all the unconfirmed coins, excluded coins and dust.
+        if coin_record.coin.name() in unconfirmed_removals:
+            continue
+        if coin_record.coin in exclude:
+            continue
+        if coin_record.coin.amount < min_coin_amount:
+            continue
+        valid_spendable_coins.append(coin_record.coin)
+        sum_spendable_coins += coin_record.coin.amount
+
+    # This happens when we couldn't use one of the coins because it's already used
+    # but unconfirmed, and we are waiting for the change. (unconfirmed_additions)
+    if sum_spendable_coins < amount:
+        raise ValueError(
+            "Can't make this transaction at the moment. "

```

Comment:
`ValueError("Transaction for {amount} is greater than spendable balance of {sum_spendable_coins}. There may be other transactions pending")`